### PR TITLE
Fixing json parsing issues detected by libfuzz.

### PR DESCRIPTION
**Pull request by nicolasnoble**
cc @murgatroid99 & @jboeuf

### CODE REVIEW

File: src/core/json/json_reader.c [hunk 1]
```diff
@@ -280,13 +280,14 @@ grpc_json_reader_status grpc_json_reader_run(grpc_json_reader *reader) {
             break;
 
           case GRPC_JSON_STATE_OBJECT_KEY_STRING:
-            GPR_ASSERT(reader->unicode_high_surrogate == 0);
+            if (reader->unicode_high_surrogate != 0)
```
Comment (ctiller) [inline src/core/json/json_reader.c]: Any chance of a {}?

**Comment by ctiller** [comment]
Fuzzer is happy so far. Will keep it running for a bit.

File: src/core/json/json_reader.c (hunk 1 above)
Comment (jboeuf) [inline src/core/json/json_reader.c]: +1

**Comment by jboeuf** [comment]
LGTM except for the missing curlies. Thanks!

**Comment by nicolasnoble** [comment]
Interop tests broken because slave went offline, and typical c++ tsan failure.

**Comment by ctiller** [comment]
LGTM

**Comment by ctiller** [comment]
Stale?

**Pull request merged**